### PR TITLE
Get toxVer more ready

### PR DESCRIPTION
**Pull request by GrayHatter**

### CODE REVIEW

File: toxcore/tox.c [hunk 1]
```diff
@@ -72,23 +72,25 @@ typedef struct Messenger Tox;
...
 
 bool tox_version_is_compatible(uint32_t major, uint32_t minor, uint32_t patch)
 {
-    //TODO
-    return 1;
+    if (major != TOX_VERSION_MAJOR)
+        return 0;
+    else
+        return 1;
 }
```
Comment (nurupo) [inline toxcore/tox.c]:
```C
return major == TOX_VERSION_MAJOR;
```

**Comment by nurupo** [comment]
Your `tox_version_is_compatible` function has different behaviour from [`TOX_VERSION_IS_API_COMPATIBLE`](https://github.com/irungentoo/toxcore/blob/master/toxcore/tox.h#L165) macro.
Their behaviours should match.

File: toxcore/tox.c (hunk 1 above)
Comment (GrayHatter) [inline toxcore/tox.c]: it's like this because if minor is > *_VERSION_MINOR, we should fail. But that's a discussion that hasn't been resolved yet.

**Comment by nurupo** [comment]
Why was this merged without my comment addressed?
@irungentoo @GrayHatter @iphydf

**Pull request merged**